src/v/cloud_storage/remote_segment.cc

```diff
@@ -872,7 +874,73 @@ ss::future<> remote_segment::run_hydrate_bg() {
     _hydration_loop_running = false;
 }
 
-ss::future<> remote_segment::hydrate() {
+namespace {
+
+void log_hydration_abort_cause(
+  const retry_chain_logger& logger,
+  const ss::lowres_clock::time_point& deadline,
+  storage::opt_abort_source_t as) {
```

Comment:
nit: it doesn't look like this is ever null anymore?

Reply:
Yes, this can be changed to a normal reference now. I'll follow this up